src/Contracts/ArrayParameterizedRuleInterface.php

```diff
@@ -0,0 +1,22 @@
+<?php
+
+declare(strict_types=1);
+
+namespace Oshomo\CsvUtils\Contracts;
+
+interface ArrayParameterizedRuleInterface
+{
+    /**
+     * Should return an array of the allowed parameters.
+     * The allowed parameters should be tokenized string
+     * e.g :min, :max, :first, :last etc.
+     */
+    public function allowedParameters(): array;
+
+    /**
+     * Should return an array of parameter values as strings
+     * parsed in the same order and format as allowedParameters().
+     * This will aid in mapping our parameters to their placeholders.
+     */
+    public function parseParameterValues(array $parameters): array;
```

Comment:
I think this is only useful when there a single parameter is allowed. Instead of doing this, I would rather update https://github.com/hoshomoh/CSVUtils/blob/34afb02e0e2506346ce429c2d7ff64bb7eadd5d9/src/Helpers/FormatsMessages.php#L101

```
protected function replaceParameterPlaceholder(
    string $message,
    array $allowedParameters,
    array $parameters
): string {
    $hasMultipleAllowedParameter = count($allowedParameters) > 1;
    $search = $hasMultipleAllowedParameter ? $allowedParameters : $allowedParameters[0];
    $replace = $hasMultipleAllowedParameter ? $parameters : implode(',', $parameters);

    return str_replace($search, $replace, $message);
}
```

Reply:
Yes, this works as well. I just wanted to keep it open in case someone needed multiple parameters.